tests/unit/iontests.js

```diff
@@ -273,30 +273,46 @@ define([
             return fs.readFileSync(path, options);
         }
 
-        function roundTripEventStreams(reader) {
+        function roundTripEventStreams(input) {
             let streams = [];
-            streams.push(new es.IonEventStream(reader));
+            streams.push(new es.IonEventStream(new ion.makeReader(input)));
+            let textWriter = new ion.makeTextWriter();
+            textWriter.writeValues(new ion.makeReader(input));
+            textWriter.close();
+            let text = textWriter.getBytes();
+            streams.push(new es.IonEventStream(new ion.makeReader(text)));
+            let binaryWriter = new ion.makeBinaryWriter();
+            binaryWriter.writeValues(new ion.makeReader(input));
+            binaryWriter.close();
+            let binary = binaryWriter.getBytes();
+            streams.push(new es.IonEventStream(new ion.makeReader(binary)));
+
+            //second pass
+            textWriter = new ion.makeTextWriter();
+            textWriter.writeValues(new ion.makeReader(text));
+            textWriter.close();
+            streams.push(new es.IonEventStream(new ion.makeReader(textWriter.getBytes())));
 
-            let eventWriter = ion.makeTextWriter();
-            streams[0].writeEventStream(eventWriter);
-            eventWriter.close();
-            streams.push(new es.IonEventStream(new ion.makeReader(eventWriter.getBytes())));
+            binaryWriter = new ion.makeBinaryWriter();
```

Comment:
Should this have a `new`?  I thought `makeBinaryWriter` was a simple function not a constructor.

Reply:
doesnt seem necessary, I'll remove it